src/uct/api/uct.h

```diff
@@ -1078,6 +1078,43 @@ ucs_status_t uct_iface_set_am_tracer(uct_iface_h iface, uct_am_tracer_t tracer,
 
 /**
  * @ingroup UCT_RESOURCE
+ * @brief Disable synchronous progress for the interface
+ *
+ * Notify the transport that it should avoid doing anything
+ * during uct_worker_progress() and try to progress asynchronously. 
```

Comment:
I don't think we need to "try to progress asynchronously". the transport will "progress async" anyway, if it has CB_ASYNC support. Also, UCP could use WAKEUP support to re-register the progress.
Also - can this function be called from progress thread?